doc/specs/#1043 - Set the initial position of the Terminal/spec.md

```diff
@@ -0,0 +1,87 @@
+
+**initialX**: int. If useDefaultInitialPos is false, this set the initial horizontal position of the top-left corner of the window. Default is 0.
+
+**initiallY**: int. If useDefaultInitialPos is false, this set the initial vertical position of the top-left corner of the window. Default is 0.
+
+**maximizeWhenLaunch**: true or false. Determine if the window should be maximized upon launch. If true, we ignore initialCols and initialRows. Default is false.
+
+The flow chart of this process:
+
+![Sol Design](images/FlowChart.png)
+
+Step 2 to 6 should be done in AppHost::_HandleCreateWindow, which is consistent to the current code.
+
+In step 4, we may need to consider the dpi of the current monitor and multi-monitor scenario when calculating the initial position of the window.
+
+## UI/UX Design
+
+Upon successful implementation, the user is able to add new properties to the json profile file, which is illustrated in the picture below:
+
+![Sol Design](images/ProfileSnapshot.png)
+
+The rest of the UI will be the same of the current Terminal experience, except that the initial position may be different.
+
+## Capabilities
+
+### Accessibility
+
+This feature will not impact accessibility of Windows Terminal.
+
+### Security
+
+This should not introduce any new security issues.
+
+### Reliability
+
+This new feathre allows the users to set custom initial position of the Terminal App window, which helps them to avoid embarassing window launch situation such as launching outside the screen bounds. Thus, it improves the reliability.
+
+### Compatibility
+
+This feature won't break existing features of Terminal.
+
+### Performance, Power, and Efficiency
+
+More data reading and calculation will be included in Terminal Launch process, which may inversely influence the launch time. However, the impact is trivial. 
+
+## Potential Issues
+
+1. The dpi of the monitor may impact the initial position even if the json profile settings are the same. We need to consider the dpi of the user's monitor and calculate the initial position in the current screen coordinates. Testing with different monitor dpi are also necessary.
+
+2. We need to consider multi-monitor scenario. If the user has multiple monitors, we must guarantee that the Terminal could be iniitalized as expected. More discussions on what should we do in this scenario with the github community is needed. 
+
+## Future considerations
+
+For now, this feature only allows the user to set initial positon and choose whether to maximize the window when launch. In the future, we may consider follow-up features like:
+
+1. Save the position of the Terminal on exit, and restore the position on the next launch. This could be a true/false feature that users could choose to set.
+
+2. We may need to consider multiple Terminal windows scenario. If the user opens multiple Terminal windows, then we need to consider how to save and restore the position. 
+

```

Comment:
One thing to also consider probably as a part of this design is supporting launching straight into not only maximized, but also full-screen mode. While we don't support full-screen mode yet, we will by 1.0, and there have already been asks to add support for launching into fullscreen mode by default. 

Having two boolean settings `maximizeWhenLaunched` and `fullscreenWhenLaunched` seems wrong to me - what if they're both true? What if we'd want to add an additional `minimizeWhenLaunched`?

Reply:
What if we add a settings called "launchMode", and have "default", "maximize" as its property for now. We may also add "minimize", "fullscreen" in the future ? I wonder if these property values are too difficult for users to remember ?

Reply:
Let's summon the wordsmiths and see what input they have @bitcrazed @cinnamon-msft

Reply:
changed